From 3af2e9fc73e13d1fdff23e0c2de99c4a539eea28 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Tue, 4 Feb 2025 14:58:48 +0100
Subject: [PATCH 1587/1695] drivers: stm32_exti: predate line in use

Ignore the RIF configuration and set as secure any used EXTI line.

For RIF aware devices, this should be replaced by a test that the
EXTI line is set as secure in the initial platform configuration.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 86a66d616829721543691904a625667cc094afc4 drivers: stm32_exti: predate line in use]
Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I9e64dc1a4b64ab7547c2d0834cf382a7d28f890f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/434933
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/stm32_exti.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/core/drivers/stm32_exti.c b/core/drivers/stm32_exti.c
index 1185a4be4..63e3bf566 100644
--- a/core/drivers/stm32_exti.c
+++ b/core/drivers/stm32_exti.c
@@ -775,6 +775,13 @@ stm32_exti_dt_get_chip_cb(struct dt_pargs *pargs, void *priv_data,
 	if (stm32_exti_event_is_configurable(exti, exti_line))
 		stm32_exti_set_type(exti, exti_line, type);
 
+	/*
+	 * Without RIF, predate the line by setting it as secure.
+	 * TODO: with RIF, check the permission
+	 */
+	if (!IS_ENABLED(CFG_STM32_RIF) || !stm32_exti_maxcid(exti))
+		stm32_exti_set_tz(exti, exti_line);
+
 	return TEE_SUCCESS;
 }
 
-- 
2.47.2

